nucypher/policy/conditions/ecdsa.py

```diff
@@ -143,14 +148,21 @@ def __init__(
         message: Any,
         signature: str,
         verifying_key: str,
+        curve: Optional[str] = NIST192p.name,
         condition_type: str = CONDITION_TYPE,
         name: Optional[str] = None,
     ):
+        if curve not in self.SUPPORTED_CURVES:
+            raise InvalidCondition(
+                f"Unsupported curve: {curve}. Supported curves are: {list(self.SUPPORTED_CURVES.keys())}"
+            )
+
         try:
             self.execution_call = ECDSAVerificationCall(
                 message=message,
                 signature=signature,
                 verifying_key=verifying_key,
+                curve=self.SUPPORTED_CURVES[curve],
```

Comment:
I think you can just pass the curve value here - see comment above. It will be validated by the Schema.